server/auth/types/authentication_type.ts

```diff
@@ -160,7 +160,7 @@ export abstract class AuthenticationType implements IAuthenticationType {
 
       // extend session expiration time
       if (this.config.session.keepalive) {
-        cookie!.expiryTime = Date.now() + this.config.session.ttl;
+        cookie!.expiryTime = Math.max(Date.now() + this.config.session.ttl, cookie.expiryTime || 0);
```

Comment:
so the value cannot be set below 1000 (the default)?

Reply:
Yes - the default is actually here: https://github.com/opensearch-project/security-dashboards-plugin/blob/ab2b85c6c5d0665edddc6d074055ed277c1567cd/server/index.ts#L78, but the idea of this block is to not shorten the existing expiry of the cookie (which seemed to occur with IDPs, since their expiry may be longer than the default of 1 hour)